test/unit/ui/marker.test.js

```diff
@@ -16,31 +16,60 @@ function createMap() {
 }
 
 test('Marker', (t) => {
+
     t.test('constructor', (t) => {
-        const el = window.document.createElement('div');
-        const marker = new Marker(el);
-        t.ok(marker.getElement(), 'marker element is created');
+        const marker = new Marker();
+        t.ok(marker, 'marker is created');
         t.end();
     });
 
-    t.test('default marker', (t) => {
+    t.test('default marker without element', (t) => {
         const marker = new Marker();
-        t.ok(marker.getElement(), 'default marker is created');
-        t.ok(marker.getOffset().equals(new Point(0, -14)), 'default marker with no offset uses default marker offset');
+        t.ok(marker.getElement(), 'marker returns default element');
+        t.end();
+    });
+
+    t.test('default marker with element', (t) => {
+        const marker = new Marker(window.document.createElement('div'));
+        t.ok(marker.getElement(), 'marker with custom element is created');
+        t.end();
+    });
+
+    t.test('default marker with anchor', (t) => {
+        const marker = new Marker();
+        t.ok(marker.getElement(), 'marker returns default element');
+        t.ok(marker.getAnchor(), 'marker returns default  anchor option');
+        t.end();
+    });
+
+    t.test('default marker with offset', (t) => {
+        const marker = new Marker();
+        t.ok(marker.getElement(), 'marker returns default element');
+        t.ok(marker.getOffset(), 'marker returns default offset option');
+        t.end();
+    });
+
+    t.test('default marker with anchor and offset', (t) => {
+        const marker = new Marker();
+        t.ok(marker.getElement(), 'marker returns default element');
+        t.ok(marker.getAnchor(), 'marker returns default  anchor option');
+        t.ok(marker.getOffset(), 'marker returns default offset option');
         t.end();
     });
 
     t.test('default marker with some options', (t) => {
-        const marker = new Marker(null, { foo: 'bar' });
+        const marker = new Marker(null, { anchor: 'bottom',  foo: 'bar' });
         t.ok(marker.getElement(), 'default marker is created');
-        t.ok(marker.getOffset().equals(new Point(0, -14)), 'default marker with no offset uses default marker offset');
+        t.ok(marker.getAnchor(), 'marker set with anchor options');
+        t.ok(marker.getOffset(), 'default marker with no offset uses default marker offset');
         t.end();
     });
 
     t.test('default marker with custom offest', (t) => {
-        const marker = new Marker(null, { offset: [1, 2] });
+        const marker = new Marker(window.document.createElement('div'), { offset: [1, 2] });
```

Comment:
default marker in this test means when no element is passed, let's leave the original test in place and you can add new tests for the new anchor functionality.